src/compiler/YamlCompiler.ts

```diff
@@ -0,0 +1,152 @@
+import {CompilationResult, ICompiler} from "./interfaces";
+import {cerr, formatDuration, GREEN_CHECK} from "../utils";
+import * as cpx from "cpx";
+import * as rimraf from "rimraf";
+import * as path from "path";
+import {exec} from "child_process";
+
+export class YamlCompiler implements ICompiler {
```

Comment:
Similar to my concerns in the other file, this is not a "yaml compiler". It is a "OpenAPI yaml compiler", and that should be reflected in the name, i.e. "OpenApiYamlCompiler". If yaml is the only supported format for OpenAPI, you may shorten to "OpenApiCompiler".